UI/Controller.js

```diff
@@ -0,0 +1,280 @@
+            post.photoLink = document.getElementById("photo").value;
+            post.descriprion = document.getElementById("decr").value;
+            post.createdAt = new Date();
+            post.likes = [];
+            post.hashtags = [];
+            let tags = document.getElementById("hash").value;
+            post.hashtags = tags.split(' ');
+            if (!Posts.addPhotoPost(post)) {
+                document.getElementById('add-photoPost').close();
+                return false;
+            }
+            if (top1 + 1 <= 10) {
+                ViewPhoto._showPost(Posts.getPhotoPost(post.id));
+                top1++;
+            }
+            document.getElementById('btn-add').close();
+            Posts.save();
+            return true;
+        }
+    }
+
+    function downLoadMore(event) {
+        if (event.target.id === "down-more") {
+            skip1 = top1;
+            let lenPosts = Posts.getLenghtPosts() - skip1;
+            top1 += 10;
+            if (lenPosts < 10) {
+                top1 -= 10;
+                top1 += lenPosts;
+            }
+
+            let morePosts = Posts.getDownMore(skip1, top1);
+            if (morePosts.length != 0) {
+                ViewPhoto.showPosts(Posts.getDownMore(skip1, top1));
+            }
+        }
+    }
+
+    function doLike(IdPost) {
+        let flag = Posts.addLikeToPost(IdPost, user);
+        
+        ViewPhoto.doLike(IdPost, flag);
+        Posts.save();
+    }
+
+    function editPosts(id) {
+        let oldPost = Posts.getPhotoPost(id);
+        const editPost = document.getElementById('dia-edit');
+        ViewPhoto.pageEditPost(oldPost, user);
+
+        if (editPost.open != true)
+            editPost.showModal();
+    }
+
+    function saveEdit (id)
+    {
+        let newPost =
+        {
+            descriprion: '',
+            photoLink: '',
+            hashtags: []
+        }
+        newPost.photoLink = document.getElementById('new-photo').value;
+        newPost.descriprion = document.getElementById('new-decr').value;
+        newPost.hashtags = document.getElementById('new-hash').value.split(',');
+
+        if (!Posts.editPhotoPost(id, newPost)) {
+            return false;
+        }
+        ViewPhoto._editPost(id, Posts.getPhotoPost(id));
+        document.getElementById('dia-edit').close();
+        Posts.save();
+        
+    }
+
+    function doDel(id) {
+          
+        if (!Posts.removePhotoPost(id)) {
+            return false;
+        }
+        skip1 -= 1;
+        top1 -= 1;
+        ViewPhoto._removePost(id);
+      
+        Posts.save();
+        return true;
+    }
+
+    function eventPost() {
+        const postsContaier = document.querySelector('main');
+        postsContaier.addEventListener('click', (event) => {
+            const target = event.target.closest('[data-action]');
+            if (event.target.id == 'like' || event.target.id == 'edit' || event.target.id == 'delete'
```

Comment:
why do we check this ?
Doesn't data-action attribute say us that click was on the actionable button and doesn't it says which one of ections exactly is need to be performed?